src/main/java/com/synopsys/integration/blackduck/installer/dockerswarm/edit/AlertLocalOverridesEditor.java

```diff
@@ -85,18 +111,26 @@ public void edit(File installDirectory) throws BlackDuckInstallerException {
         addEnvironmentVariable(ymlBuilder, "ALERT_PROVIDER_BLACKDUCK_BLACKDUCK_API_KEY", alertBlackDuckInstallOptions.getBlackDuckApiToken());
         addEnvironmentVariable(ymlBuilder, "ALERT_PROVIDER_BLACKDUCK_BLACKDUCK_TIMEOUT", alertBlackDuckInstallOptions.getBlackDuckTimeoutInSeconds());
 
-        if (!alertEncryption.isEmpty() || !customCertificate.isEmpty()) {
-            appendAlertSecrets(ymlBuilder, alertEncryption, customCertificate);
-            appendSecrets(ymlBuilder, alertEncryption, customCertificate);
+        if (!alertEncryption.isEmpty() || !customCertificate.isEmpty() || alertDatabase.hasSecrets()) {
+            appendAlertSecrets(ymlBuilder, alertEncryption, customCertificate, alertDatabase);
+            appendSecrets(ymlBuilder, alertEncryption, customCertificate, alertDatabase);
         }
 
-        try (Writer writer = new FileWriter(configFile.getFileToEdit(), true)) {
+        // write the file clean because the original has an alertdb service which may not be there if it's an external deployment.
+        // Also the original file has environment variables set for postgres but if secrets are used those environment variables cannot be present.
+        // To cover all the deployment options the safest thing to do is write the file clean.
+        try (Writer writer = new FileWriter(configFile.getFileToEdit(), false)) {
```

Comment:
@ekerwin  was there a reason that you didn't overwrite the file originally?

Reply:
Yeah, we don't want to replace any of the commented out code that users could edit themselves after the install occurs. We'll need to find a different way to handle alert's overrides to support these versions.